packages/ts/react-signals/src/FullStackSignal.ts

```diff
@@ -5,6 +5,21 @@ import { createSetStateEvent, type StateEvent } from './events.js';
 
 const ENDPOINT = 'SignalsHandler';
 
+/**
+ * The type of function that is used to define a callback that is called when
+ * the `then` method is called on an operation object.
+ */
+export type ThenCallback = () => void;
+
+/**
+ * A return type for signal operations.
```

Comment:
Nit:
This can be improved a bit to express the fact that the operation enables the user to chain then callbacks.